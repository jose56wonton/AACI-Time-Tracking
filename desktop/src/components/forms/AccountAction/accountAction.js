import React from 'react';
import PropTypes from 'prop-types';

import { withStyles } from '@material-ui/core/styles';
import { IconButton, Tooltip } from '@material-ui/core';
import { ShowChart, Check, Close, ArrowBack, Storage } from '@material-ui/icons';

import styles from './styles';

const AccountActionForm = props => {
  const { classes, isWorking, clockIn, clockOut, analyze, back } = props;
  return (
    <div className={classes.hero}>
      <div className={classes.heroContent}>
        {isWorking ? (
          <Tooltip open={true} title="Clock Out" classes={{ tooltip: classes.toolTip }} placement="bottom">
            <IconButton onClick={clockOut} className={classes.button}>
              <Close className={classes.buttonIcon} />
            </IconButton>
          </Tooltip>
        ) : (
            <Tooltip open={true} title="Clock In" classes={{ tooltip: classes.toolTip }} placement="bottom">
              <IconButton onClick={clockIn} className={classes.button}>
                <Check className={classes.buttonIcon} />
              </IconButton>
            </Tooltip>
          )}
        <Tooltip open={true} title="Export" classes={{ tooltip: classes.toolTip }} placement="bottom">
          <IconButton onClick={props.export} className={classes.button}>
            <Storage className={classes.buttonIcon} />
          </IconButton>
        </Tooltip>
        <Tooltip open={true} title="Analysis" classes={{ tooltip: classes.toolTip }} placement="bottom">
          <IconButton onClick={analyze} className={classes.button}>
            <ShowChart className={classes.buttonIcon} />
          </IconButton>
        </Tooltip>
        <Tooltip open={true} title="Back" classes={{ tooltip: classes.toolTip }} placement="bottom">
          <IconButton onClick={back} className={classes.button}>
            <ArrowBack className={classes.buttonIcon} />
          </IconButton>
        </Tooltip>

      </div>
    </div>
  );
};

AccountActionForm.propTypes = {
  classes: PropTypes.object.isRequired,
  back: PropTypes.func.isRequired,
  clockIn: PropTypes.func.isRequired,
  isWorking: PropTypes.number,
  clockOut: PropTypes.func.isRequired,
  export: PropTypes.func.isRequired,
  analyze: PropTypes.func.isRequired
};

export default withStyles(styles)(AccountActionForm);
