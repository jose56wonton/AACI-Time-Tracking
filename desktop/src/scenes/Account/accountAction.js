import React, { Component } from 'react';
import PropTypes from 'prop-types';

import AccountActionContainer from 'containers/Account/accountActionContainer';

class AccountAction extends Component {
  render() {
    return <AccountActionContainer type={this.props.type} />;
  }
}

AccountAction.propTypes = {
  type: PropTypes.string.isRequired
};

export default AccountAction;
