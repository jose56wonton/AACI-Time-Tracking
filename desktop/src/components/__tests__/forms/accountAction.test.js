import React from 'react';
import { shallow } from 'enzyme';

import { AccountAction } from 'components/forms/AccountAction/accountAction';
import AccountActionHOC from 'components/forms/AccountAction';

const props = {  
  classes: {},
  back: jest.fn(),
  clockIn: jest.fn(),
  isWorking: 0,
  clockOut: jest.fn(),
  export: jest.fn(),
  analyze: jest.fn(),
  type: 'type'
};

const setup = () => {  
  return shallow(<AccountAction {...props} />);    
};

const setupHOC = () => {
  return shallow(<AccountActionHOC {...props} />);
};

describe('Account Action Component', () => {
  it('should render correctly', () => {
    const wrapper = setup();
    expect(wrapper).toMatchSnapshot();    
  });
  it('should render correctly withStyles', () => {
    const wrapper = setupHOC();
    expect(wrapper).toMatchSnapshot();
  });
});