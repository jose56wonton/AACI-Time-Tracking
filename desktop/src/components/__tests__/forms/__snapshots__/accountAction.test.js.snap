// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Account Action Component should render correctly 1`] = `
<div>
  <div>
    <WithStyles(Tooltip)
      classes={
        Object {
          "tooltip": undefined,
        }
      }
      open={true}
      placement="bottom"
      title="Clock In"
    >
      <WithStyles(IconButton)
        onClick={[Function]}
      >
        <pure(CheckIcon) />
      </WithStyles(IconButton)>
    </WithStyles(Tooltip)>
    <WithStyles(Tooltip)
      classes={
        Object {
          "tooltip": undefined,
        }
      }
      open={true}
      placement="bottom"
      title="Back"
    >
      <WithStyles(IconButton)
        onClick={[Function]}
      >
        <pure(ArrowBackIcon) />
      </WithStyles(IconButton)>
    </WithStyles(Tooltip)>
  </div>
</div>
`;

exports[`Account Action Component should render correctly withStyles 1`] = `
<Component
  analyze={[Function]}
  back={[Function]}
  classes={
    Object {
      "button": "Component-button-3",
      "buttonIcon": "Component-buttonIcon-4",
      "hero": "Component-hero-1",
      "heroContent": "Component-heroContent-2",
      "toolTip": "Component-toolTip-5",
    }
  }
  clockIn={[Function]}
  clockOut={[Function]}
  export={[Function]}
  isWorking={0}
  type="type"
/>
`;
